controller/src/main/java/org/jboss/as/controller/extension/ExtensionRegistry.java

```diff
@@ -184,11 +184,11 @@ public Builder withHostControllerInfoAccessor(RuntimeHostControllerInfoAccessor
 
         /**
          * Overrides the default stability level of the extension registry.
-         * @param stability a stability level
+         * @param stabilitySupplier a supplier returning the stability level
          * @return a reference to this builder
          */
-        public Builder withStability(Stability stability) {
-            this.stability = stability;
+        public Builder withStabilitySupplier(Supplier<Stability> stabilitySupplier) {
```

Comment:
Better, I think, to leave the existing method in place and add a new one that accepts a provided value.

Reply:
It's a builder - why would it matter how it is set?  The existing method would just delegate to the new method using a constant supplier.

Reply:
@kabir It's in a release, so we shouldn't incompatibly change it unless we really need to:

https://github.com/wildfly/wildfly-core/blob/23.0.1.Final/controller/src/main/java/org/jboss/as/controller/extension/ExtensionRegistry.java#L182-L190